client/me/purchases/remove-purchase/index.jsx

```diff
@@ -176,28 +355,273 @@ const RemovePurchase = React.createClass( {
+				id="too_hard_input"
+				value={ this.state.questionOneText }
+				onChange={ this.handleTextOne }
+				placeholder={ this.translate( 'Where did you run into problems?' ) } />
+		);
+		reasons.too_hard = (
+			<FormLabel key="too_hard">
+				<FormRadio
+					name="too_hard"
+					value="too_hard"
+					checked={ 'too_hard' === this.state.questionOneRadio }
+					onChange={ this.handleRadioOne } />
+				<span>{ this.translate( 'It was too hard to set up my site.' ) }</span>
+				{ 'too_hard' === this.state.questionOneRadio ? too_hard_input : null }
+			</FormLabel>
+		);
+
+		const did_not_include_input = (
+			<FormTextInput
+				className="remove-purchase__reason-input"
+				name="did_not_include_input"
+				id="did_not_include_input"
+				value={ this.state.questionOneText }
+				onChange={ this.handleTextOne }
+				placeholder={ this.translate( 'What are we missing that you need?' ) } />
+		);
+		reasons.did_not_include = (
+			<FormLabel key="did_not_include">
+				<FormRadio
+					name="did_not_include"
+					value="did_not_include"
+					checked={ 'did_not_include' === this.state.questionOneRadio }
+					onChange={ this.handleRadioOne } />
+				<span>{ this.translate( 'This upgrade didn\'t include what I needed.' ) }</span>
+				{ 'did_not_include' === this.state.questionOneRadio ? did_not_include_input : null }
+			</FormLabel>
+		);
+
+		const only_need_free_input = (
+			<FormTextInput
+				className="remove-purchase__reason-input"
+				name="only_need_free_input"
+				id="only_need_free_input"
+				value={ this.state.questionOneText }
+				onChange={ this.handleTextOne }
+				placeholder={ this.translate( 'Is there anything we can do to improve our upgrades?' ) } />
+		);
+		reasons.only_need_free = (
+			<FormLabel key="only_need_free">
+				<FormRadio
+					name="only_need_free"
+					value="only_need_free"
+					checked={ 'only_need_free' === this.state.questionOneRadio }
+					onChange={ this.handleRadioOne } />
+				<span>{ this.translate( 'All I need is the free plan.' ) }</span>
+				{ 'only_need_free' === this.state.questionOneRadio ? only_need_free_input : null }
+			</FormLabel>
+		);
+
+		const another_reason_one_input = (
+			<FormTextInput
+				className="remove-purchase__reason-input"
+				name="another_reason_one_input"
+				value={ this.state.questionOneText }
+				onChange={ this.handleTextOne }
+				id="another_reason_one_input" />
+		);
+		reasons.another_reason_one = (
+			<FormLabel key="another_reason_one">
+				<FormRadio
+					name="another_reason_one"
+					value="another_reason_one"
+					checked={ 'another_reason_one' === this.state.questionOneRadio }
+					onChange={ this.handleRadioOne } />
+				<span>{ this.translate( 'Another reason…' ) }</span>
+				{ 'another_reason_one' === this.state.questionOneRadio ? another_reason_one_input : null }
+			</FormLabel>
+		);
+
+		for ( let i in this.state.questionOneOrder ) {
+			ordered_reasons.push( reasons[ this.state.questionOneOrder[ i ] ] );
```

Comment:
using map would be a little cleaner here... no mutations, more functional, javascript hipstery, etc. Like so:
```js
const orderedReasons = this.state.questionOneOrder.map( question => reasons[ questions ] );
```
I'm using a fat arrow function with an implicit return. Actually, to make it really clean we'd extract the order via destructuring, like:
```js
const { questionOneOrder } = this.state;
const orderedReasons = questionOneOrder.map( question => reasons[ question ] );
```